packages/frontend/src/pages/TokenSwap/utils/hooks/usePools.js

```diff
@@ -0,0 +1,34 @@
+import { useMemo } from 'react';
+import { useSelector } from 'react-redux';
+
+import { selectAllPools, selectPoolsLoading } from '../../../../redux/slices/swap';
+
+const retrievePools = (allPools, keys) => {
+    for (const key of keys) {
+        const pools = allPools[key];
+
+        if (pools && Object.keys(pools).length) {
+            return pools;
+        }
+    }
+
+    return null;
+};
+
+export default function usePools({ tokenIn, tokenOut }) {
+    const allPools = useSelector(selectAllPools);
+    const poolsLoading = useSelector(selectPoolsLoading);
+
+    const pools = useMemo(() => {
```

Comment:
Lets add a comment about what is happening there and why.
Also, should we add `error` & `isInitialized` here?
It may be useful to show error in case we cant load them and to distinct between "no pools available" and "didn't load yet"